Detectors/Calibration/testMacros/getRunParameters.cxx

```diff
@@ -0,0 +1,197 @@
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+// executable to get the interaction rate and duration of a run from CCDB
+
+#include <fstream>
+#include <stdio.h>
+#if !defined(__CLING__) || defined(__ROOTCLING__)
+#include "CCDB/BasicCCDBManager.h"
+#include "CommonDataFormat/InteractionRecord.h"
+#include "CCDB/CcdbApi.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "DataFormatsCTP/Scalers.h"
+#include "DataFormatsCTP/Configuration.h"
+#include "DataFormatsParameters/GRPMagField.h"
+#include "CommonTypes/Units.h"
+#include <boost/program_options.hpp>
+#endif
+
+using namespace o2::ctp;
+namespace bpo = boost::program_options;
+
+const double orbitDuration = 88.924596234; // us
+
+void writeIRtoFile(float ir)
+{
+
+  FILE* fptr = fopen("IR.txt", "w");
+  if (fptr == NULL) {
+    printf("ERROR: Could not open file to write IR!");
+    return;
+  }
+  fprintf(fptr, "%.2f", ir);
+  fclose(fptr);
+}
+
+void writeDurationToFile(long duration)
+{
+
+  FILE* fptr = fopen("Duration.txt", "w");
+  if (fptr == NULL) {
+    printf("ERROR: Could not open file to write IR!");
+    return;
+  }
+  fprintf(fptr, "%ld", duration);
+  fclose(fptr);
+}
+
+void writeBFieldToFile(float b)
+{
+
+  FILE* fptr = fopen("BField.txt", "w");
+  if (fptr == NULL) {
+    printf("ERROR: Could not open file to write IR!");
+    return;
+  }
+  fprintf(fptr, "%.2f", b);
+  fclose(fptr);
+}
+
+bool initOptionsAndParse(bpo::options_description& options, int argc, char* argv[], bpo::variables_map& vm)
+{
+  options.add_options()(
+    "run,r", bpo::value<int>()->default_value(8), "Run number to inspect")(
+    "enable-debug,d", bpo::value<bool>()->default_value(false)->implicit_value(true), "Enable debug logs")(
+    "help,h", "Produce help message.");
+
+  try {
+    bpo::store(parse_command_line(argc, argv, options), vm);
+
+    // help
+    if (vm.count("help")) {
+      std::cout << options << std::endl;
+      return false;
+    }
+
+    bpo::notify(vm);
+  } catch (const bpo::error& e) {
+    std::cerr << e.what() << "\n\n";
+    std::cerr << "Error parsing command line arguments; Available options:\n";
+
+    std::cerr << options << std::endl;
+    return false;
+  }
+  return true;
+}
+
+int main(int argc, char* argv[])
+{
+  bpo::options_description options("Allowed options");
+  bpo::variables_map vm;
+  if (!initOptionsAndParse(options, argc, argv, vm)) {
+    return -1;
+  }
+
+  auto run = vm["run"].as<int>();
+  auto debug = vm["enable-debug"].as<bool>();
+
+  float ir = 0.f;
+  long duration = 0;
+  // duration as O2end - O2start:
+  auto& ccdb_inst = o2::ccdb::BasicCCDBManager::instance();
+  ccdb_inst.setURL("https://alice-ccdb.cern.ch");
```

Comment:
Should not the server be configurable?

Reply:
I would not do it configurable, because these values come from the official CCDB. They are used for official reco.

Reply:
Cooked up values can be passed to the reco directly, see:
https://github.com/AliceO2Group/O2DPG/blob/378cb4fc4b73737f71b78f0223bab12b89126f16/DATA/production/configurations/asyncReco/setenv_extra.sh#L159